server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/SetMessagesCreationProcessor.java

```diff
@@ -107,11 +109,7 @@ public SetMessagesResponse process(SetMessagesRequest request, MailboxSession ma
     private void handleCreate(CreationMessageEntry create, Builder responseBuilder, MailboxSession mailboxSession) {
         try {
             validateIsUserOwnerOfMailboxes(create, mailboxSession);
-            validateImplementedFeature(create, mailboxSession);
-            validateArguments(create, mailboxSession);
-            MessageWithId created = handleOutboxMessages(create, mailboxSession);
-            responseBuilder.created(created.getCreationId(), created.getValue());
-
+            performCreate(create, responseBuilder, mailboxSession);
```

Comment:
handleCreate is calling performCreate.
I think that bad naming scheme reflect the lack of proper abstraction. What was wrong with this version ?

Reply:
I did not like the double check to the feature:
 - 1 check to see if it's draft OR outbox, 1 second check to perform the feature (that was always sending).

Generally, error management is performed in a separated method. Is that a bad thing? I tend to concider separating these concerns is good.